### DISCUSSION TITLE
Custom BuildEventArgs after BinaryFormatter Removal

### DISCUSSION THREAD

**Discussion by Forgind** [dotnet/msbuild#7582] [General]
With .NET 7, [BinaryFormatter will no longer be available](https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#net-7-nov-2022) for dotnet-org repositories to use. We currently serialize custom BuildEventArgs using BinaryFormatter. If you or someone you know use that functionality, we want to know! Post below with a little information about your scenario including how hard it would be to accomplish your goal without making your own type.

The use of BinaryFormatter will be removed either way, but this will guide whether we try to provide any means of working around its absence versus just pulling it out. Let us know your thoughts!

**Comment by RomanWoerner** [comment]


**Reply by benvillalobos** [reply]
It sounds like you're having an issue using BinaryFormatter at runtime, but this discussion is about removing BF from MSBuild itself 🙂

**Comment by JanKrivanek** [comment]
Closing as BinaryFormatter removal work is already done (https://github.com/dotnet/msbuild/issues/6215).
Specifically subclassing of BuildEventArgs will lead o build time errors (after #8917) from net8 RC1 and VS17.9 P2. Possible users are guided to use built-in `Extended*EventArgs`
